### PR TITLE
Voice message button: Show proper disabled color.

### PR DESCRIPTION
**Pull request by julioromano**
Also changes a bit our own IconButton api to allow to mirror material's and allow color customization.

### CODE REVIEW

**Comment by julioromano** [comment]
Screenshots: https://github.com/vector-im/element-x-android/actions/runs/6691203075

**Comment by julioromano** [comment]
Assigning to @jmartinesp for the change to the designsystem module and to @jonnyandrew for voice messages

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/IconButton.kt [hunk 1]
```diff
@@ -38,13 +39,13 @@ fun IconButton(
     onClick: () -> Unit,
     modifier: Modifier = Modifier,
     enabled: Boolean = true,
+    colors: IconButtonColors = IconButtonDefaults.iconButtonColors(
```
Comment (jmartinesp) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/IconButton.kt:42]: Technically you could have used a `LocalCompositionProvider` and set the `LocalContentColor` to `iconSecondary` too. But I think it makes sense to be able to customize these here too.

**Review by jmartinesp — APPROVED** [review]
LGTM design-system wise.

**Pull request merged**